libraries/bot-azure/src/main/java/com/microsoft/bot/azure/blobs/BlobsTranscriptStore.java

```diff
@@ -0,0 +1,423 @@
+                               setFrom(from);
+                               setRecipient(recipient);
+                               setLocale(activityAndBlob.getLeft().getLocale());
+                               setLocalTimestamp(activityAndBlob.getLeft().getTimestamp());
+                               setTimestamp(activityAndBlob.getLeft().getTimestamp());
+                               setChannelId(activityAndBlob.getLeft().getChannelId());
+                               setConversation(activityAndBlob.getLeft().getConversation());
+                               setServiceUrl(activityAndBlob.getLeft().getServiceUrl());
+                               setReplyToId(activityAndBlob.getLeft().getReplyToId());
+                           }
+                       };
+
+                       logActivityToBlobClient(tombstonedActivity, activityAndBlob.getRight(), true).join();
+                   }
+                });
+
+                return CompletableFuture.completedFuture(null);
+            default:
+                String blobName = this.getBlobName(activity);
+                BlobClient blobClient = containerClient.getBlobClient(blobName);
+                logActivityToBlobClient(activity, blobClient, true).join();
+                return CompletableFuture.completedFuture(null);
+        }
+    }
+
+    public CompletableFuture<PagedResult<Activity>> getTranscriptActivities(String channelId, String conversationId,
+                                                                            @Nullable String continuationToken,
+                                                                            OffsetDateTime startDate) {
+        if (startDate == null) {
+            startDate = OffsetDateTime.now(ZoneId.of("UTC"));
+        }
+
+        final int pageSize = 20;
+
+        if (StringUtils.isBlank(channelId)) {
+            throw new IllegalArgumentException("Missing channelId");
+        }
+
+        if (StringUtils.isBlank(conversationId)) {
+            throw new IllegalArgumentException("Missing conversationId");
+        }
+
+        PagedResult<Activity> pagedResult = new PagedResult<Activity>();
+
+        String token = null;
+        List<BlobItem> blobs = new ArrayList<BlobItem>();
+        do {
+            String prefix = String.format("%1$s/%2$s/", sanitizeKey(channelId), sanitizeKey(conversationId));
+            Iterable<PagedResponse<BlobItem>> resultSegment = containerClient.listBlobsByHierarchy(prefix)
+                .iterableByPage(token);
+            token = null;
+
+            for (PagedResponse<BlobItem> blobPage: resultSegment) {
+                for (BlobItem blobItem: blobPage.getValue()) {
+                    DateTime parseDateTime = DateTime.parse(blobItem.getMetadata().get("Timestamp"));
+                    Instant instantTime = Instant.ofEpochMilli(parseDateTime.getMillis());
+                    OffsetDateTime offsetParserDateTime = OffsetDateTime.ofInstant(instantTime, ZoneId.of("UTC"));
```

Comment:
Not quite sure about this, I think that you need to check how we implemented this in Timex